### DISCUSSION TITLE
best way to debug and track the source code

### DISCUSSION THREAD

**Discussion by FxamarAboali** [schwabe/ics-openvpn#1753] [General]
What is the best way to debug and track the source code and know the data flow during the data communication with OpenVPN server?

**Comment by schwabe** [comment]
Do not use a complicated Android app to do that where debugging and doing all that stuff is much harder. Use normal OpenVPN on Linux.

**Reply by FxamarAboali** [reply]
what is the best way to debugging OpenVPN on Linux
